mm2src/mm2_main/src/lp_swap/swap_v2_common.rs

```diff
@@ -0,0 +1,364 @@
+    let swaps_ctx = SwapsContext::from_ctx(ctx).expect("SwapsContext::from_ctx should not fail");
+    let db = swaps_ctx.swap_db().await?;
+    let transaction = db.transaction().await?;
+
+    let table = transaction.table::<SavedSwapTable>().await?;
+    let saved_swap_json = match table.get_item_by_unique_index("uuid", id).await? {
+        Some((_item_id, SavedSwapTable { saved_swap, .. })) => saved_swap,
+        None => return MmError::err(SwapStateMachineError::NoSwapWithUuid(id)),
+    };
+
+    let swap_repr = serde_json::from_value(saved_swap_json)?;
+    Ok(swap_repr)
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub(super) async fn get_unfinished_swaps_uuids(
+    ctx: MmArc,
+    swap_type: u8,
+) -> MmResult<Vec<Uuid>, SwapStateMachineError> {
+    async_blocking(move || {
+        select_unfinished_swaps_uuids(&ctx.sqlite_connection(), swap_type)
+            .map_to_mm(|e| SwapStateMachineError::StorageError(e.to_string()))
+    })
+    .await
+}
+
+#[cfg(target_arch = "wasm32")]
+pub(super) async fn get_unfinished_swaps_uuids(
+    ctx: MmArc,
+    swap_type: u8,
+) -> MmResult<Vec<Uuid>, SwapStateMachineError> {
+    let index = MultiIndex::new(IS_FINISHED_SWAP_TYPE_INDEX)
+        .with_value(BoolAsInt::new(false))?
+        .with_value(swap_type)?;
+
+    let swaps_ctx = SwapsContext::from_ctx(&ctx).expect("SwapsContext::from_ctx should not fail");
+    let db = swaps_ctx.swap_db().await?;
+    let transaction = db.transaction().await?;
+    let table = transaction.table::<MySwapsFiltersTable>().await?;
+    let table_items = table.get_items_by_multi_index(index).await?;
+
+    Ok(table_items.into_iter().map(|(_item_id, item)| item.uuid).collect())
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub(super) async fn mark_swap_as_finished(ctx: MmArc, id: Uuid) -> MmResult<(), SwapStateMachineError> {
+    async_blocking(move || Ok(set_swap_is_finished(&ctx.sqlite_connection(), &id.to_string())?)).await
+}
+
+#[cfg(target_arch = "wasm32")]
+pub(super) async fn mark_swap_as_finished(ctx: MmArc, id: Uuid) -> MmResult<(), SwapStateMachineError> {
+    let swaps_ctx = SwapsContext::from_ctx(&ctx).expect("SwapsContext::from_ctx should not fail");
+    let db = swaps_ctx.swap_db().await?;
+    let transaction = db.transaction().await?;
+    let table = transaction.table::<MySwapsFiltersTable>().await?;
+    let mut item = match table.get_item_by_unique_index("uuid", id).await? {
+        Some((_item_id, item)) => item,
+        None => return MmError::err(SwapStateMachineError::NoSwapWithUuid(id)),
+    };
+    item.is_finished = true.into();
+    table.replace_item_by_unique_index("uuid", id, &item).await?;
+    Ok(())
+}
+
+pub(super) fn init_additional_context_impl(ctx: &MmArc, swap_info: ActiveSwapV2Info) {
+    subscribe_to_topic(ctx, swap_v2_topic(&swap_info.uuid));
+    let swap_ctx = SwapsContext::from_ctx(ctx).expect("SwapsContext::from_ctx should not fail");
+    swap_ctx.init_msg_v2_store(swap_info.uuid, bits256::default());
```

Comment:
I am just wondering why was `accept_only_from` needed in legacy swap implementation?